src/software/thunderscope/robot_communication.py

```diff
@@ -78,6 +75,19 @@ def __init__(
             target=self.__run_primitive_set, daemon=True
         )
 
+        # map of robot id to the individual control mode
+        self.robot_control_mode_dict: dict[int, IndividualRobotMode] = {
+            robot_id: IndividualRobotMode.NONE
+            for robot_id in list(range(0, NUM_ROBOTS))
+        }
+
+        # map of robot id to the number of times to send a estop.
+        # each robot has it's own separate countdown
+        # TODO come up with better name
+        self.robot_estop_send_countdown_latches: dict[int, int] = {
+            rid: 0 for rid in range(NUM_ROBOTS)
+        }
```

Comment:
I can change to use any constant such as `MAX_ROBOT_IDS_PER_SIDE` but what about `DIV_A_NUM_ROBOTS`? - we could extend and shouldn't have the problem at all that you are describing?

I flip-flopped a lot between using a `list` and a `dict`, and just prefer using the `dict` for clarity and no duplicates. I will refactor to be similar to the `robot_view.py` process - should I abstract the two into a single function that is used in both? Lastly, since, we have a `NONE control mode, then I thought it is a great default value for init on all robots, which is why I went with using the `dict`.